### PR TITLE
[CI:DOCS] Update release notes for Podman v4.0.0

### PR DESCRIPTION
**Pull request by mheon**
As the title says.

Our largest release so far by a significant margin, in terms of release notes (I think 2.0 probably has us beat in terms of diffstat, but I haven't verified that).

### CODE REVIEW

**Comment by mheon** [comment]
@containers/podman-maintainers PTAL

**Comment by mheon** [comment]
For the curious:

v3.4.4 to v4.0.0 is:
 1142 files changed, 27628 insertions(+), 21416 deletions(-)
v2.2.1 to v3.0.0 is:
 703 files changed, 30388 insertions(+), 21267 deletions(-)
v1.9.3 to v2.0.0 is: 
 880 files changed, 34425 insertions(+), 36600 deletions(-)

All figures exclude changes to `vendor/` (which isn't strictly fair here, because a lot of Paul's work landed in c/common, but thinking of a regex to allow vendor/github.com/containers and nothing else is complicated and my brain is fried.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,5 +1,255 @@
...
+- The `--net` option to `podman create`, `podman pod create`, `podman run`, and `podman play kube` can now, when specifying a network name, also specify advanced network options such as `alias`, `ip`, `mac`, and `interface_name`, allowing advanced configuration of networks when creating containers connected to more than one network.
+- The `podman play kube` command can now specify the `--net` option multiple times, to connect created containers and pods to multiple networks.
+- The `podman create`, `podman pod create`, and `podman run` commands now support a new option, `--ipv6`, to specify a static IPv6 address for the created container or pod to use.
+- Macvlan networks can now configure the mode of the network via the `-o mode=` option.
+- When using the CNI network stack, a new network driver, `ipvlan`, is now available.
+- In support of the new support for switching network backends, the `podman info` command will now print the network backend in use (Netavark or CNI), and the backend to use can be specified via `containers.conf`. Please note that it is not recommended to switch backends while containers exist, and a system reboot is recommended after doing so.
```
Comment (rhatdan) [inline RELEASE_NOTES.md:12]: This sentence is difficult to parse.

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,5 +1,255 @@
...
+- The `:O` option for volumes, which specifies that an overlay filesystem will be mounted over the volume and ensures changes do not persist, is now supported with named volumes as well as bind mounts.
+- Podman containers created from a user-specified root filesystem (via `--rootfs`) can now create an overlay filesystem atop the user-specified rootfs which ensures changes will not persist by suffixing the user-specified root filesystem with `:O`.
+- The `podman save` command has a new option, `--uncompressed`, which saves the layers of the image without compression ([#11613](https://github.com/containers/podman/issues/11613)).
+- Podman supports a new log driver for containers, `passthrough`, which logs all output directly to the STDOUT and STDERR of the `podman` command; it is intended for use in systemd-managed containers.
+- The `podman build` command now supports two new options, `--unserenv` and `--all-platforms`.
```
Comment (rhatdan) [inline RELEASE_NOTES.md:41]: s/unsetrenv/unsetenv/g

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -2026,7 +2276,7 @@
 - Updated Buildah to v1.7, picking up a number of bugfixes
 - Updated containers/image library to v1.5, picking up a number of bugfixes and performance improvements to pushing images
 - Updated containers/storage library to v1.10, picking up a number of bugfixes
-- Work on the remote Podman client for interacting with Podman remotely over Varlink is progressing steadily, and many image and pod commands are supported - please see the [Readme](https://github.com/containers/podman/blob/main/remote_client.md) for details
+- Work on the remote Podman client for interacting with Podman remotely over Varlink is progressing steadily, and many image and pod commands are supported - please see the [Readme](https://github.com/containers/podman/blob/master/remote_client.md) for details
```
Comment (flouthoc) [inline RELEASE_NOTES.md:2279]: I think this still has to point to `main` not `master`

**Review by vrothberg — COMMENTED** [review]
Besides the other comments, LGTM

Great work, @mheon !

**Comment by mheon** [comment]
Force pushed with fixes

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -1,5 +1,256 @@
...
+- The `--no-trunc` argument to the `podman search` command now defaults to true.
+- Rootless port forwarding using the `rootlessport` port forwarder is now handled by a separate binary, not Podman itself, which results in significantly reduced memory usage ([#10790](https://github.com/containers/podman/issues/10790)).
+- The `podman system connection ls` command now has a separate output column to show which connection is currently the default (instead appending `*` to the default connection's name) ([#12019](https://github.com/containers/podman/issues/12019)).
+- The `--kernel-memory` option to `podman run` and `podman create` has been deprecated in the upstream OCI runtime specification, and is now also deprecated in Podman and will be removed in a future release. Use of the flag will result in a warning.
+- Podman will now ship build the pause image used by pods locally, instead of pulling it from the network (using the existing `catatoinit` binary used for `podman run --init`). This allows pods to be easily used on systems without an internet connection.
```
Comment (TomSweeneyRedHat) [inline RELEASE_NOTES.md:89]: nit for later "ship build" to one or the other

File: RELEASE_NOTES.md [hunk 5]
```diff
@@ -1,5 +1,256 @@
...
+- The `podman image sign` command now supports a new option, `--authfile`, to specify an alternative path to authentication credentials ([#10866](https://github.com/containers/podman/issues/10866)).
+- The `podman load` command now supports downloading files via HTTP and HTTPS if a URL is given ([#11970](https://github.com/containers/podman/issues/11970)).
+- The `podman push` command now supports a new option, `--compression-format`, to choose the compression algorithm used to compress image layers.
+- The `podman volume create` command now allows volumes using the `local` driver that require mounting to be used by non-root users. This allows `tmpfs` and `bind` volumes to be created by non-root users ([#12013](https://github.com/containers/podman/issues/12013)).
+- A new command, `podman dial-stdio`, has been added; this command should not be invoked directly, but is used by some clients of the Docker Remote API, and is provided for Docker compatibility ([#11668](https://github.com/containers/podman/issues/11668)).
+
+### Breaking Changes
```
Comment (TomSweeneyRedHat) [inline RELEASE_NOTES.md:66]: I would lead with this.

**Pull request merged**